test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpLogExporterTests.cs

```diff
@@ -25,51 +28,162 @@ namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests
             otlpLogRecord = logRecord.ToOtlpLog();
             Assert.NotNull(otlpLogRecord);
             Assert.Equal("Hello from tomato 2.99.", otlpLogRecord.Body.StringValue);
         }
+
+        [Fact]
+        public void CheckToOtlpLogRecordTraceIdSpanIdFlagWithDroppedActivity()
+        {
+            this.logger.LogInformation("Log within a dropped activity");
+            var logRecord = this.exportedItems[0];
+            var otlpLogRecord = logRecord.ToOtlpLog();
+
+            Assert.Null(Activity.Current);
+            Assert.True(otlpLogRecord.TraceId.IsEmpty);
+            Assert.True(otlpLogRecord.SpanId.IsEmpty);
+            Assert.True(otlpLogRecord.Flags == 0);
+        }
+
+        [Fact]
+        public void CheckToOtlpLogRecordSpanIdTraceIdFlag()
+        {
+            // preparation
+            var sampler = new AlwaysOnSampler();
+            var exportedActivityList = new List<Activity>();
+            var activitySourceName = "toOtlpLogRecordTest";
+            var activitySource = new ActivitySource(activitySourceName);
+            using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+                .AddSource(activitySourceName)
+                .SetSampler(sampler)
+                .AddInMemoryExporter(exportedActivityList)
+                .Build();
+
+            using var activity = activitySource.StartActivity("Activity");
+
+            // execution
+            this.logger.LogInformation("Log within activity marked as RecordOnly");
+
+            // assertion
+            var logRecord = this.exportedItems[0];
+            var otlpLogRecord = logRecord.ToOtlpLog();
+
+            var currentActivity = Activity.Current;
+            Assert.NotNull(Activity.Current);
+
+            var expectedTraceId = currentActivity.TraceId;
+            var expectedSpanId = currentActivity.SpanId;
+            Assert.Equal(expectedTraceId.ToString(), ActivityTraceId.CreateFromBytes(otlpLogRecord.TraceId.ToByteArray()).ToString());
+            Assert.Equal(expectedSpanId.ToString(), ActivitySpanId.CreateFromBytes(otlpLogRecord.SpanId.ToByteArray()).ToString());
+            Assert.Equal((uint)logRecord.TraceFlags, otlpLogRecord.Flags);
+        }
+
+        [Fact]
+        public void CheckToOtlpLogRecordSeverityText()
+        {
+            this.options.IncludeFormattedMessage = true;
+
+            this.logger.LogInformation("Hello from {name} {price}.", "tomato", 2.99);
+            Assert.Single(this.exportedItems);
+
+            var logRecord = this.exportedItems[0];
+            var otlpLogRecord = logRecord.ToOtlpLog();
+
+            Assert.NotNull(otlpLogRecord);
+            Assert.Equal(logRecord.LogLevel.ToString(), otlpLogRecord.SeverityText);
+        }
+
+        [Fact]
+        public void CheckToOtlpLogRecordFormattedMessage()
+        {
+            this.options.IncludeFormattedMessage = true;
+
+            this.logger.LogInformation("OpenTelemetry!");
```

Comment:
I like the enthusiasm.

Reply:
Or anger? 😆

Reply:
`OpenTelemetry?!` 😠

Reply:
I'd prefer to use something like the usual Hello from {food} {price}, for testing formatted message. (i.e which has string template, which gets formatted with actual food and price). Or that could be in addition to this one :)